client/src/search.tsx

```diff
@@ -65,7 +65,9 @@ function useSearchIndex(): readonly [
     const gather = async () => {
```

Comment:
Reading through this function (in the context of the rest of these changes) I'm not entirely clear if the intent is that we "return fast" - not waiting for the collection items fetch to return - in which case this doesn't need to be async, or if we want to wait - in which case I'm not sure some of the changes in `client/src/plus/collections-quicksearch.ts` putting `collectionItems` in the module scope are necessary - since we can just await the fetch operation.

Reply:
The module level cache is the happy path if the search component gets reloaded and nothing changed. This is perceivable performance on prod as the fetch can take a while.

Reply:
I had somehow missed that we save this in localstorage - this makes sense.